src/main/java/edu/rpi/legup/ui/proofeditorui/treeview/TreeView.java

```diff
@@ -1,756 +1,881 @@
+        if (view.getType() == NODE) {
+            TreeNodeView nodeView = (TreeNodeView) view;
+            nodeView.getParentView().setChildView(null);
+        }
+        else {
+            TreeTransitionView transitionView = (TreeTransitionView) view;
+            transitionView.getParentViews().forEach((TreeNodeView n) -> n.removeChildrenView(transitionView));
+        }
+    }
+
+    /**
+     * When the edu.rpi.legup.user hovers over the transition, draws the corresponding rules image
+     *
+     * @param g the graphics to use to draw
+     */
+    public void drawMouseOver(Graphics2D g) {
+        if (selection.getHover().getType() == TRANSITION && ((TreeTransitionView) selection.getHover()).getTreeElement().isJustified()) {
+            TreeTransition transition = (TreeTransition) selection.getHover().treeElement;
+            int imgWidth = 100;
+            int imgHeight = 100;
+
+            BufferedImage image = new BufferedImage(imgWidth, imgHeight, BufferedImage.TYPE_INT_ARGB);
+            image.createGraphics().drawImage(transition.getRule().getImageIcon().getImage(), 0, 0, null);
+            Point mousePoint = selection.getMousePoint();
+            g.drawImage(image, mousePoint.x, mousePoint.y - 50, imgWidth, imgHeight, null);
+        }
+    }
+
+    public void resetView() {
+        this.tree = null;
+        this.rootNodeView = null;
+        this.selection.clearSelection();
+        this.selection.clearHover();
+    }
+
+    /**
+     * Called when a tree puzzleElement is added to the tree
+     *
+     * @param treeElement TreeElement that was added to the tree
+     */
+    @Override
+    public void onTreeElementAdded(TreeElement treeElement) {
+        if (treeElement.getType() == NODE) {
+            addTreeNode((TreeNode) treeElement);
+        }
+        else {
+            addTreeTransition((TreeTransition) treeElement);
+        }
+        repaint();
+    }
+
+    /**
+     * Called when a tree puzzleElement is removed from the tree
+     *
+     * @param element TreeElement that was removed to the tree
+     */
+    @Override
+    public void onTreeElementRemoved(TreeElement element) {
+        if (element.getType() == NODE) {
+            TreeNode node = (TreeNode) element;
+            TreeNodeView nodeView = (TreeNodeView) viewMap.get(node);
+
+            nodeView.getParentView().setChildView(null);
+            removeTreeNode(node);
+        }
+        else {
+            TreeTransition trans = (TreeTransition) element;
+            TreeTransitionView transView = (TreeTransitionView) viewMap.get(trans);
+
+            //unlock ancestor elements if case rule deleted
+            Rule rule = trans.getRule();
+            for(TreeNode node : trans.getParents()){
+                //only if the last case of a case rule will be deleted
+                if (rule instanceof CaseRule && node.getChildren().isEmpty()) {
+                    //System.out.println("TRANSISTION UNLOCKING");
```

Comment:
Remove commented code before submitting changes (even if they're only for debugging).

Reply:
removed